### PR TITLE
Add channel-aware checklist item resolution to Playbooks MCP

### PR DESCRIPTION
**Pull request by Willyfrog**
## Summary

When a user asks the agent to "mark task X done" via the Playbooks MCP, `check_item` required the caller to already know the exact `run_id` and zero-based `checklist_number`/`item_number`. Nothing steered the model to discover those, and there was no by-name matching — so it guessed indexes or failed.

This adds discovery, makes `check_item` self-correcting, and extends the same bounds-check pattern across all index-based checklist/section tools.

### Discovery (commit 1)
- **New tool `resolve_channel_context`** — given a `channel_id`, lists that channel's runs with each checklist item formatted as `[checklist_number][item_number] title (state)`, so the model can go straight to `check_item`. This is the "current context resolver."
- **New tool `find_checklist_item`** — matches a `query` against item titles, searching **channel-first** (explicit `run_id` → the given `channel_id`'s in-progress runs → the user's in-progress runs), and returns the `run_id` + indexes. Skips closed/skipped items unless `include_closed` is set. Handles 1 / many / no matches.
- **`list_runs`** — accepts an optional `channel_id` filter (the primitive the resolvers build on; the `GET runs` API already supports it with a channel-view permission check).

### Bounds-checking across all index-based tools (commit 2)
Every index-based item/section tool now fetches the run and validates the indexes against its real checklists, returning the actual items on a miss instead of an opaque API error — and, for destructive tools, never deleting the wrong thing.
- Shared helpers: `fetchRunForBounds`, `checkChecklistIndex`, `checkItemIndex`, `checkInsertPosition` (the last allows `position == count` for move appends).
- Applied to `check_item`, `add_checklist_item`, `set_checklist_item_due_date`, `set_checklist_item_assignee`, `remove_checklist_item`, `move_checklist_item`, `edit_checklist_item`, `rename_section`, `remove_section`, `move_section`.
- `check_item` also no-ops when the item is already in the requested state; `remove_*` name the removed item/section in the result.
- Resolve-first pointers added to the mutating tool descriptions.

### Design note / known limitation
The Playbooks MCP server only receives the **user ID** today (`mcphelper` extracts `X-Mattermost-UserID` and nothing else). No channel context reaches the tool handlers, so `channel_id` must be passed explicitly by the calling model from its conversation context. Making channel-first behavior fully automatic would require an upstream Agents + `mcphelper` change (forward an `X-Mattermost-ChannelID` header + a `GetChannelID` accessor) — scoped as a follow-up, not included here.

Per review, `check_item` stays index-based (resolver-first) rather than also accepting a title to self-resolve; that convenience is a possible later follow-up.

## Testing
- Unit tests for both new tools and for the bounds-check/no-op paths (out-of-range listing, destructive no-delete, move append-position edge cases, closed-item filtering, run-id scoping, query validation).
- Existing table tests updated with run fixtures for the new fetch-first behavior; provider registration test asserts both new tools register end-to-end via the MCP server.
- `go test ./internal/playbooksmcp/...` and `go test ./server/ -run MCP` pass; `golangci-lint` clean on the package.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/playbooksmcp/tools/checklist.go [hunk 1]
```diff
@@ -378,6 +484,17 @@ func toolMoveChecklistItem(ctx context.Context, client APIClient, args MoveCheck
 		return "", err
 	}
 
+	run, err := fetchRunForBounds(ctx, client, args.RunID)
+	if err != nil {
+		return "", err
+	}
+	if err := checkItemIndex(run, args.SourceChecklistIdx, args.SourceItemIdx, "source_item_idx"); err != nil {
```
Comment (JulienTant) [inline internal/playbooksmcp/tools/checklist.go]: `checkItemIndex` hardcodes "checklist_number" for the checklist existence check, but `MoveChecklistItemArgs` has no `checklist_number` field (it uses `source_checklist_idx`). An out of range `source_checklist_idx` here will produce an error telling the caller to fix `checklist_number`, which doesn't exist on this tool.

**Review by JulienTant — CHANGES_REQUESTED** [review]
I think my pending comment is blocking

I started an AI review and it reported this that I find worthy to consider:

> 6 of 10 newly bounds-checked tools have zero adversarial (out-of-range) test coverage (test-engineer) — add_checklist_item, set_checklist_item_due_date, edit_checklist_item, set_checklist_item_assignee, remove_checklist_item, rename_section only got their fixture enlarged so the new bounds checks never trip; the bounds-check calls could be deleted from these 6 tools today without any test failing.

**Comment by Willyfrog** [comment]
@coderabbitai please, do a full review

File: internal/playbooksmcp/tools/checklist.go [hunk 2]
```diff
@@ -177,7 +192,69 @@ func toolCheckItem(ctx context.Context, client APIClient, args CheckItemArgs) (s
...
+}
+
+// fetchRunForBounds retrieves the run so callers can validate indexes against
+// its real checklists before mutating, giving actionable errors on a miss.
+func fetchRunForBounds(ctx context.Context, client APIClient, runID string) (playbookRunDetail, error) {
+	var run playbookRunDetail
+	if err := client.Get(ctx, fmt.Sprintf("runs/%s", runID), nil, &run); err != nil {
+		return playbookRunDetail{}, fmt.Errorf("failed to get run: %w", err)
+	}
+	return run, nil
```
[the remark this answers is not in the document]
Comment (Willyfrog) [reply internal/playbooksmcp/tools/checklist.go:214]: not implementable, and not a correctness defect

**Review by JulienTant — APPROVED** [review]
